x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -8,24 +8,57 @@ import * as React from 'react';
 import ReactDOM from 'react-dom';
 import { CoreStart, AppMountParameters } from 'kibana/public';
 import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
+import { History, createHashHistory } from 'history';
+import { Route, Router, Switch } from 'react-router-dom';
 
 /**
  * This module will be loaded asynchronously to reduce the bundle size of your plugin's main bundle.
  */
 export function renderApp(coreStart: CoreStart, { element }: AppMountParameters) {
   coreStart.http.get('/api/endpoint/hello-world');
 
-  ReactDOM.render(<AppRoot />, element);
+  const history = createHashHistory();
+  ReactDOM.render(<AppRoot history={history} />, element);
 
   return () => {
     ReactDOM.unmountComponentAtNode(element);
   };
 }
 
-const AppRoot = React.memo(() => (
+interface RouterProps {
+  history: History;
+}
+
+const AppRoot: React.FC<RouterProps> = React.memo(({ history }) => (
```

Comment:
FYI - just noticed in the TS typing definition that `React.FC` is marked as `@deprecated` and we should use `FunctionComponent` instead (at least in the version of React that is being used in the `feature-ingest` branch 🙂 )

Reply:
good catch, fixed 👍